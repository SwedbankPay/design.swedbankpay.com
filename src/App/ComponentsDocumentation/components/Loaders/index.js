import React from "react";

import { ComponentPreview, DocContainer } from "@docutils";
import CodeTags from "@components/CodeTags";

const BasicUsage = () => (
    <>
        <h2 id="basic-usage">Basic usage</h2>
        <p>
            To use a default loader add <CodeTags type="secondary" code=".loader" /> to a <CodeTags type="primary" code="<div>" />.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="loader" />
        </ComponentPreview>
    </>
);

const CenteredLoader = () => (
    <>
        <h2 id="centered-loader">Centered loader</h2>
        <p>
            To center the loader add class <CodeTags type="secondary" code=".loader-center" />.
            <ComponentPreview language="html" showCasePanel codeFigure>
                <div className="loader loader-center" />
            </ComponentPreview>
        </p>
    </>
);

const Sizes = () => (
    <>
        <h2 id="sizes">Sizes</h2>
        <p>In addition to <CodeTags type="secondary" code=".loader" /> you can use <CodeTags type="secondary" code=".loader-sm" /> and <CodeTags type="secondary" code=".loader-lg" /> to decrease or increase the size of the loader.</p>
        <h3>Small</h3>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="loader loader-sm" />
        </ComponentPreview>
        <h3>Large</h3>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="loader loader-lg" />
        </ComponentPreview>
    </>
);

const MutedLoader = () => (
    <>
        <h2 id="muted-loader">Muted loader</h2>
        <p>
            Use class <CodeTags type="secondary" code=".loader-muted" /> to make the loader muted.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="loader loader-muted" />
        </ComponentPreview>
    </>
);

const Loaders = () => (
    <DocContainer docToc>
        <p className="lead">
            Use loaders to indicate to the user that your web application is loading.
        </p>
        <BasicUsage />
        <CenteredLoader />
        <Sizes />
        <MutedLoader />
    </DocContainer>
);

export default Loaders;

/* For testing */
export { BasicUsage, CenteredLoader, Sizes, MutedLoader };
