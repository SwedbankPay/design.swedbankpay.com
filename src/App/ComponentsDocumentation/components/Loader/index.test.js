import React from "react";
import renderer from "react-test-renderer";

import Loaders, { Overview } from "./index";

describe("Documentation: Loaders", () => {
	it("is defined", () => {
		expect(Loaders).toBeDefined();
	});

	it.skip("renders", () => {
		const wrapper = renderer.create(<Loaders />);

		expect(wrapper.toJSON()).toMatchSnapshot();
	});

	describe("Overview", () => {
		it("is defined", () => {
			expect(Overview).toBeDefined();
		});

		it.skip("renders", () => {
			const wrapper = renderer.create(<Overview />);

			expect(wrapper.toJSON()).toMatchSnapshot();
		});
	});
});
