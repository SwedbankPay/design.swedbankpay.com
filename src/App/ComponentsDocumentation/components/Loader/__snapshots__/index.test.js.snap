// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation: Loaders Overview renders 1`] = `
<section
  id="loader-page-overview"
>
  <h2
    id="overview"
  >
    Overview
  </h2>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeOuterTag={true}
    showCasePanel={true}
    showCasePanelAdvanced={
      Object {
        "elements": Array [
          Object {
            "component": <Loader />,
            "description": "Loading spinners should be used when the wait time is anticipated to extend over one second.",
            "options": Object {
              "radio": Array [
                Object {
                  "id": "color_radio",
                  "title": "Color",
                  "values": Array [
                    Object {
                      "name": "Default",
                    },
                    Object {
                      "name": "Inverted",
                      "value": Object {
                        "backgroundDark": true,
                        "inverted": true,
                      },
                    },
                  ],
                },
                Object {
                  "id": "size_radio",
                  "title": "Size",
                  "values": Array [
                    Object {
                      "name": "Default",
                      "value": Object {
                        "size": "",
                      },
                    },
                    Object {
                      "name": "Small",
                      "value": Object {
                        "size": "loader-sm",
                      },
                    },
                  ],
                },
              ],
            },
            "title": "Loader",
          },
        ],
        "id": "no-tabs",
      }
    }
  />
  <section
    id="loaders-how-to-use"
  >
    <h3>
      How to use loader in context
    </h3>
    <div
      className="row"
    >
      <div
        className="col-12"
        key="0"
      >
        <div
          className="slab slab-plain slab-border-success"
          key="0"
        >
          <span
            className="h3"
          >
            Do
          </span>
          <div
            className="d-flex justify-content-center px-3 h-100 align-items-center pb-5"
          >
            <div
              className="d-flex flex-column align-items-center col-5"
            >
              <p
                className="text-align-center"
              >
                <b>
                  Opening BankID app on your mobile device, please don't close this window
                </b>
              </p>
              <div
                className="loader"
              />
            </div>
          </div>
        </div>
      </div>
      <p
        className="m-3"
      >
        If the anticipated waiting time is long, itâ€™s recommended to show a text explaining what is happening to the user.
      </p>
      <div
        className="col-6"
        key="0"
      >
        <div
          className="slab slab-plain slab-border-success"
          key="0"
        >
          <span
            className="h3"
          >
            Do
          </span>
          <div
            className="d-flex justify-content-center px-3 h-100 align-items-center pb-5"
          >
            <button
              className="btn btn-primary disabled"
            >
              ID authentication
              <div
                className="loader loader-sm"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        className="col-6"
        key="1"
      >
        <div
          className="slab slab-plain slab-border-error"
          key="1"
        >
          <span
            className="h3"
          >
            Don't
          </span>
          <div
            className="d-flex justify-content-center px-3 h-100 align-items-center pb-5"
          >
            <div
              className="d-flex"
            >
              <button
                className="btn btn-primary btn-bank-id"
                disabled={true}
              >
                ID authentication
                <i
                  className="bank-id bank-id-se"
                />
              </button>
              <div
                className="loader loader-sm align-self-center"
              />
            </div>
          </div>
        </div>
      </div>
      <p
        className="m-3"
      >
        If the loading is contextual to a certain button on the page, the 
        <Link
          to="/components/buttons"
        >
          button
        </Link>
         loading state can be used instead of the loader component.
      </p>
    </div>
  </section>
</section>
`;

exports[`Documentation: Loaders renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    The loader component should be used when the site or application is performing computation, notifying   the user that their action is being processed.
  </p>
  <Overview />
</DocContainer>
`;
